src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadInstallType.cs

```diff
@@ -27,25 +29,17 @@ public static class WorkloadInstallType
         /// </summary>
         /// <param name="sdkFeatureBand">The SDK version to check.</param>
         /// <returns>The <see cref="InstallType"/> associated with the SDK.</returns>
-        public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand, string dotnetDir)
-        {
-            string installerTypePath = Path.Combine(dotnetDir, "metadata",
-                "workloads", $"{sdkFeatureBand.ToStringWithoutPrerelease()}", "installertype");
-
-            if (File.Exists(Path.Combine(installerTypePath, "msi")))
-            {
-                return InstallType.Msi;
-            }
-
-            return InstallType.FileBased;
-        }
+        public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand, string? dotnetDir) =>
+            dotnetDir is not null && File.Exists(Path.Combine(
+                dotnetDir, "metadata", "workloads", $"{sdkFeatureBand.ToStringWithoutPrerelease()}", "installertype", "msi"))
+            ? InstallType.Msi : InstallType.FileBased;
 
-        public static string GetInstallStateFolder(SdkFeatureBand sdkFeatureBand, string dotnetDir)
+        public static string GetInstallStateFolder(SdkFeatureBand sdkFeatureBand, string? dotnetDir)
         {
             var installType = GetWorkloadInstallType(sdkFeatureBand, dotnetDir);
             var architecture = RuntimeInformation.ProcessArchitecture.ToString();
 
-            if (installType == InstallType.FileBased)
+            if (dotnetDir is not null && installType == InstallType.FileBased)
```

Comment:
I think dotnetDir is explicitly non-null here?

Reply:
According to the call stack, the `SdkDirectoryWorkloadManifestProvider` has the `_sdkOrUserLocalPath` field which is nullable. So that gets passed into this as `dotnetDir`. Unless you change the design of all of these classes to be `string` vs `string?`, then you can't really undo this stuff.